homeassistant/components/sensor/recorder.py

```diff
@@ -92,6 +104,19 @@ def _time_weighted_average(
     return accumulated / (end - start).total_seconds()
 
 
+def _normalize(state: State, device_class: str) -> float:
+    """Normalize units."""
+    fstate = float(state.state)
+    if device_class not in UNIT_CONVERSIONS:
+        return fstate
+
+    unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+    if unit not in UNIT_CONVERSIONS[device_class]:
+        return fstate
```

Comment:
Should we warn and/or throw if the unit is not supported?

Reply:
Warn. Makes no sense to throw as it's in our own code.